java/flight/flight-jdbc/src/main/java/org/apache/arrow/jdbc/FlightStatement.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.jdbc;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.sql.SQLWarning;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.arrow.flight.CallOption;
+import org.apache.arrow.flight.CallOptions;
+import org.apache.arrow.flight.FlightClient;
+import org.apache.arrow.flight.FlightStream;
+import org.apache.arrow.flight.Location;
+import org.apache.arrow.flight.Ticket;
+import org.apache.arrow.memory.RootAllocator;
+
+/**
+ * Statement.
+ */
+public class FlightStatement implements java.sql.Statement {
+
+  protected final FlightConnection flightConnection;
+
+  public FlightStatement(FlightConnection flightConnection) {
+    this.flightConnection = flightConnection;
+  }
+
+  @Override
+  public ResultSet executeQuery(String query) throws SQLException {
+
+    FlightClient client = FlightClient.builder()
+        .allocator(new RootAllocator(Long.MAX_VALUE))
+        .location(Location.forGrpcInsecure(flightConnection.host, flightConnection.port))
+        .build();
+
+    CallOption callOptions = CallOptions.timeout(5, TimeUnit.SECONDS);
```

Comment:
Note that this timeout is for the entire RPC, i.e. a query that takes more than 5 seconds (to stream all the data or something) would get cancelled. Unfortunately gRPC doesn't make it easy to have per-message timeouts.